pkg/matcher/annotation_matcher.go

```diff
@@ -26,7 +26,7 @@ const (
 
 func branchMatch(prunBranch, baseBranch string) bool {
 	// if target is refs/heads/.. and base is without ref (for pullRequest)
-	if strings.HasPrefix(prunBranch, "refs/heads") && !strings.Contains(baseBranch, "/") {
+	if strings.HasPrefix(prunBranch, "refs/heads") && !strings.Contains(baseBranch, "refs/heads") {
```

Comment:
i feel like, this is more verbose

```
// if target is refs/heads/.. and base is without ref (for pullRequest)
	if strings.HasPrefix(prunBranch, "refs/heads/") {
		ref := baseBranch
		if !strings.HasPrefix(baseBranch, "refs/heads/") {
			ref = "refs/heads/" + baseBranch
		}
		g := glob.MustCompile(prunBranch)
		if g.Match(ref) {
			return true
		}
	} else {
		prunRef := prunBranch
		if !strings.HasPrefix(prunBranch, "refs/heads/") {
			prunRef = "refs/heads/" + prunBranch
		}
		g := glob.MustCompile(prunRef)
		if g.Match(baseBranch) {
			return true
		}
	}

Reply:
:laughing: 
but i feel its more clear with the existing code as well

I mean its understandable

Reply:
the code in PR handled two case and this cover all 4

also the second condition should be also prefix not contains

Reply:
Even current code handles all scenario isn't it :thinking: 

> also the second condition should be also prefix not contains

ya having prefix make sense

Reply:
the `if` condition which is removed, was handling the two cases, which are not getting handled now, was left for the last compile glob, to avoid any regression it is safe side two handle those here.

Reply:
It will handle by compile glob there won't be any issues

But anyway updated code :+1: